### ISSUE TITLE
Dust notes & planner iterations

### ISSUE THREAD

**Issue by grod220** [penumbra-zone/web#1205]
Reported by @plaidfinch

Currently, in the planner we balance with fees in a loop. If it exceeds 100 iterations, we throw an error: https://github.com/penumbra-zone/web/blob/e86448e295a7fe4aeaee3d0e6d3d9478e41933ac/packages/wasm/crate/src/planner.rs#L512-L514

That means the planner will fail when an account has many small "dust" notes. A remediation has been the `pcli sweep` method to consolidate notes. I suppose the error was added as a kind of timeout if there were too many iterations.

Options:
- Remove entirely
- Increase the iteration count significantly
- Make a clock-based timeout

**Comment by zbuc** [comment]
For usability it might be nice if there was some kind of heuristic to detect dust notes and recommend the user sweep.

**Comment by hdevalence** [comment]
> Remove entirely

We probably don't want this, it's important for planning to terminate.

> Increase the iteration count significantly

We probably don't want this, since making a tx with 100 spends will be very expensive in the web context.

> Make a clock-based timeout

By the time we are in this situation we have the problem that there are a lot of dust notes, the error is just us noticing that, so I'm not sure this helps.

**Comment by hdevalence** [comment]
An alternative option is that we could have Prax auto-sweep notes as it detects them.

Currently the pcli logic sweeps 8 notes at a time. We could do something similar, but automatically:

- Periodically check if there are more than 8 notes per (asset, account) pair
- Silently generate a tx that sweeps them into a single note, 8 at a time, and broadcast it

The downside is that we're performing an action without explicit user consent. Also if it's automatic there might be a potential deanonymizatikn vector (i send dust notes to an address then watch you submit a transaction and now i know you control it)

For now I think that doing nothing is a safe choice.

**Comment by zbuc** [comment]
> I think the heuristic is the note count right? I don't see a scenario where we hit that condition where we wouldn't want to sweep.


Yes, something like "if there are more than 10 notes for a given asset ID, perform a sweep" would probably be good enough. 

I agree it's unlikely to be common on mainnet but the added complexity from the auto-sweep is low and could prevent some frustration so I think it's worth considering.

**Comment by TalDerei** [comment]
tagging https://github.com/penumbra-zone/web/issues/1595 as related
